src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -499,54 +496,64 @@ private static string GetCommandId(Dictionary<string, long> dic)
             return hash.ToString("X");
         }
 
-        internal List<(UpdateDatabaseCommand Update, string Reason)> GetUnusedAutoIndexes(DatabaseObservationState databaseState)
+        internal List<(UpdateDatabaseCommand Update, string Reason)> GetUnusedAutoIndexes(Dictionary<int, DatabaseObservationState> shardStates)
```

Comment:
Wonder if it would be nicer to split this method into 
1. `GetUnusedAutoIndexes` (up to the part where we read the settings) - per shard or non-sharded database
2.  `CleanUnusedAutoIndexes` - combine all shards or non sharded